homework1/task5/maximum_subarray/maximum_subarray.py

```diff
@@ -0,0 +1,10 @@
+from typing import List
+
+
+def find_maximal_sub_array_sum(nums: List[int], k: int) -> int:
+    set_of_sum = set()
+    for i in range(len(nums) + 1):
+        for j in range(i):
+            if len(nums[j:i]) <= k:
+                set_of_sum.add(sum(nums[j:i]))
```

Comment:
В данном случае дважды вычисляется срез списка. Можно присвоить его переменной чтобы немного упростить работу и улучшить читаемость
```suggestion
            _list = nums[j:i]
            if len(_list) <= k:
                set_of_sum.add(sum(_list))
```